plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/TableStatisticsWriter.java

```diff
@@ -112,6 +108,23 @@ public StatisticsFile writeStatisticsFile(
                 collectedStatistics);
         Map<Integer, CompactSketch> ndvSketches = collectedStatistics.ndvSketches();
 
+        return writeStatisticsFile(session, table, fileIO, snapshotId, ndvSketches);
+    }
+
+    public StatisticsFile rewriteStatisticsFile(ConnectorSession session, Table table, long snapshotId)
+    {
+        TableOperations operations = ((HasTableOperations) table).operations();
+        FileIO fileIO = operations.io();
+        // This will rewrite old statistics file as ndvSketches map is empty
+        return writeStatisticsFile(session, table, fileIO, snapshotId, Map.of());
+    }
+
+    private GenericStatisticsFile writeStatisticsFile(ConnectorSession session, Table table, FileIO fileIO, long snapshotId, Map<Integer, CompactSketch> ndvSketches)
```

Comment:
Do we need to copy the files to a new copy, or can we reference the existing files with the new snapshot in-place?

Reply:
Referencing the same file from two snapshots does not work. File will be deleted due to cleanup of expired snapshot.